Document the Recorder tool (#3063)
MicrosoftDocs/edge-developer: microsoft-edge/devtools-guide-chromium/recorder/index.md

Reviewer: rename the file from index.md to conform to 
https://review.learn.microsoft.com/en-us/help/platform/seo-optimize-urls?branch=main#name-your-files
which doesn't mention "index.md".
@captainbrosset

Reviewer: but pls also see https://review.learn.microsoft.com/en-us/help/platform/seo-optimize-urls?branch=main#url-structure
@captainbrosset

Reviewer: keeping filename "index.md".  Might ask info arch team.  I wish to get rid of the multiple instances of generic file name "index.md".

   ![The demo TODO app and the Recorder tool in DevTools, next to it](./index-images/recorder-tool.png)

<!-- ====================================================================== -->
## Record a user flow

To start recording a new user flow:

1. Click **Create a new recording**.

1. Under **Recording name**, enter a name for the recording, such as "Adding TODO tasks".

1. Click **Start recording** at the bottom of the **Recorder** tool:

   ![The recording name and start recording button](./index-images/start-recording.png)

1. In the rendered webpage, interact with the page. For example, enter "task 1" in the **Add a task** input box, and then press **Enter**. Repeat these steps multiple times.

   A red circle appears next to the recording name, indicating that the recording is in progress. The **Recorder** tool records your interactions and displays them as a list of steps:

   ![The recorded user interactions displayed as a list in the Recorder tool](./index-images/list-of-interactions.png)

> 1. To stop the recording, at the bottom of the list of recorded interactions, click **End recording**.

Reviewer: ```suggestion
1. To stop the recording, at the bottom of the list of recorded interactions, click the **End recording** button.
```
if it is a button

   The recording stops, and the circle next to the recording name turns blue, indicating that the recording is complete:

   ![The finalized recording and the Replay button](./index-images/final-recording.png)

<!-- ====================================================================== -->
## Record other user flows

> To record a new user flow, once you've stopped the recording of the previous user flow, click the **Create a new recording** (![Create a new recording button](./index-images/new-recording-icon.png)) button in the top-left corner of the **Recorder** tool, and then repeat the steps from [Record a user flow](#record-a-user-flow):

Reviewer: ```
To record a new user flow, once you've stopped the recording of the previous user flow: 

1. Click the **Create a new recording** (![Create a new recording button](./index-images/new-recording-icon.png)) button in the top-left corner of the **Recorder** tool:

   ![The Create a new recording button](./index-images/create-new-recording.png)

1. Repeat the steps from [Record a user flow](#record-a-user-flow), above.
```

> ![The Create a new recording button](./index-images/create-new-recording.png)

Reviewer: ```suggestion
![The 'Create a new recording' button](./index-images/create-new-recording.png)
```
i tested single-quote previously, usage looks ok.  didn't test screen reader

<!-- ====================================================================== -->
## Replay a user flow

You can replay user flows either from the list of recorded user flows, or when viewing the details of a given user flow.

> * To view the list of recorded user flows, click the dropdown menu that's in the **Recorder** tool toolbar, at the top, and select **2 recording(s)**. The label might show a different number of recordings, depending on how many recordings you've made.

Reviewer: ```suggestion
* To view the list of recorded user flows, click the dropdown menu that's in the **Recorder** tool toolbar, at the top, and then select **2 recording(s)**. The label might show a different number of recordings, depending on how many recordings you've made.
```
sequential steps list within sentence

Reviewer: , and then select **2 recording(s)**.

> * To view the list of recorded user flows, click the dropdown menu that's in the **Recorder** tool toolbar, at the top, and select **2 recording(s)**. The label might show a different number of recordings, depending on how many recordings you've made.

Reviewer: can we use a construction like:
Click the dropdown arrow next to **Replay** in the top-right corner of the **Recorder** tool
instead of:
, at the top"

> * To view the details of a given user flow, click the dropdown menu that's in the **Recorder** tool toolbar, at the top, and click the name of the user flow you want to view.

Reviewer: ```suggestion
* To view the details of a given user flow, click the dropdown menu that's in the **Recorder** tool toolbar, at the top, and then click the name of the user flow you want to view.
```
sequential steps list within sentence. vs simultaneous UI actions.

Reviewer: * To view the details of a given user flow, click the dropdown list that's in the **Recorder** tool toolbar, at the top, and then click the name of a user flow.

<!-- ------------------------------ -->
#### Replay a user flow from the list of recordings

> 1. Click **Play recording** next to the recording that you want to replay:

Reviewer: ```suggestion
1. Click the **Play recording** button next to the recording that you want to replay:
```

>    ![The list of recordings, and the play button next to each recording](./index-images/play-from-list.png)

Reviewer: ```suggestion
   ![The list of recordings, and the Play button next to each recording](./index-images/play-from-list.png)
```

1. Wait while the **Recorder** tool replays the previously recorded user interactions. An animation of the steps that are being replayed appears in the **Recorder** tool:

   ![The replayed interactions displayed as an animation in the Recorder tool](./index-images/replayed-interactions-animation.png)

   When the replay is complete, the animation stops, and the **Recorder** tool displays the recording's steps.

<!-- ------------------------------ -->
#### Replay a user flow from a recording

> 1. Click **Replay** in the top-right corner of the **Recorder** tool:

Reviewer: ```suggestion
1. Click the **Replay** button in the top-right corner of the **Recorder** tool:
```
rendered: 
https://review.learn.microsoft.com/en-us/microsoft-edge/devtools-guide-chromium/recorder/?branch=pr-en-us-3063#replay-a-user-flow-from-a-recording

   ![The Replay button on a recorded user flow details page](./index-images/play-from-recording.png)

1. Wait while the **Recorder** tool replays the previously recorded user interactions. An animation of the steps that are being replayed appears in the **Recorder** tool:

   ![The replayed interactions displayed as an animation in the Recorder tool](./index-images/replayed-interactions-animation.png)

   When the replay is complete, the animation stops, and the **Recorder** tool displays the recording's steps.

<!-- ------------------------------ -->
#### Slowly replay a user flow to investigate issues

> To see what's happening in the rendered page while the user interactions are being replayed, slow down the replay speed. By default, the **Recorder** tool replays user flows as fast as it can. To slow down the replay speed:

Reviewer: ```suggestion
To see what's happening in the rendered page while the user interactions are being replayed, slow down the replay speed. By default, the **Recorder** tool replays user flows as fast as it can.

 To slow down the replay speed:
```
lead-in for list of steps is left-justified

> 1. Click the dropdown arrow next to **Replay** in the top-right corner of the **Recorder** tool, and then select a speed from the dropdown menu, such as **Slow**, **Very slow**, or **Extremely slow**:

Reviewer: ```suggestion
*  Click the dropdown arrow next to **Replay** in the top-right corner of the **Recorder** tool, and then select a speed from the dropdown menu, such as **Slow**, **Very slow**, or **Extremely slow**:
```
* check the number of space chars in list
* unordered list if 1 item, per style guide or contrib Markup guide.

   ![The Replay button and the replay speed dropdown menu](./index-images/replay-slow.png)

<!-- ------------------------------ -->

> #### Pause while replaying a user flow by using step breakpoints

Reviewer: ```suggestion
#### Pause while replaying a user flow by using Step breakpoints
```
would capzn work to indicate a type of breakpoint?  check article  
https://learn.microsoft.com/en-us/microsoft-edge/devtools-guide-chromium/javascript/breakpoints

Reviewer: #### Pause while replaying a user flow, by using breakpoints

> To inspect the rendered webpage at a specific point of the user flow, pause the replay by using a breakpoint. To add a breakpoint to a user flow:

Reviewer: ```suggestion
To inspect the rendered webpage at a specific point of the user flow, pause the replay by using a breakpoint.

To add a breakpoint to a user flow:
```

> 1. Click the **Open step actions** (![The Open step actions icon](./index-images/step-actions-icon.png)) icon next to the step you want to pause at, and then select **Add breakpoint**:

Reviewer: ```suggestion
1. Click the **Open step actions** (![The Open step actions icon](./index-images/step-actions-icon.png)) button next to the step you want to pause at, and then select **Add breakpoint**:
```
a button contains an icon as its label

>     ![The Open step actions icon and the Add breakpoint button](./index-images/add-breakpoint.png)

Reviewer: ```suggestion
    ![The 'Open step actions' button and the Add breakpoint button](./index-images/add-breakpoint.png)
```

Reviewer: ```
![The 'Open step actions' button and the 'Add breakpoint' menu item](./index-images/add-breakpoint.png)
```

> 1. To start the user flow, click **Replay** in the top-right corner of the **Recorder** tool. The replay pauses when it reaches the step with the breakpoint. The **Continue** and **Execute one step** buttons also become available in the toolbar:

Reviewer: ```suggestion
1. To start the user flow, click **Replay** in the top-right corner of the **Recorder** tool.

    The replay pauses when it reaches the step with the breakpoint. The **Continue** and **Execute one step** buttons also become available in the toolbar:
```

Reviewer: ```
1. To start the user flow, click **Replay** in the top-right corner of the **Recorder** tool.

   The replay pauses when it reaches the step that has a breakpoint. The **Continue** and **Execute one step** buttons also become available in the toolbar:
```

    ![The paused replay at a step with a breakpoint](./index-images/paused-replay.png)

> 1. Click the **Execute one step** button to execute the next step and pause again, or click the **Continue** button to continue the replay until the next breakpoint, or the end of the user flow, is reached.

Reviewer: ```suggestion
1. Click the **Execute one step** button to execute the next step and pause again, or click the **Continue** button to continue the replay until the next breakpoint (or the end of the user flow) is reached.
```
better continuity of flow

  * To change the value of a selector, under **Selectors**, click the input box that contains the value of the selector, and then enter the new value of the selector:

    ![The selector value input box](./index-images/edit-selector-value.png)

<!-- ------------------------------ -->
#### Delete a step

To delete a step from an existing user flow:

1. Click the **Open step actions** (![The Open step actions icon](./index-images/step-actions-icon.png)) icon next to the step you want to delete, and then select **Remove step**:

   ![The Open step actions icon and the Remove step button](./index-images/remove-step.png)

<!-- ====================================================================== -->
## Delete a user flow

To delete a previously recorded user flow, when viewing the list of recordings, click **Delete recording** (![The Delete recording icon](./index-images/delete-icon.png)) next to the recording you want to delete:

![The list of recordings, and the delete button next to each recording](./index-images/delete-recording.png)

<!-- ====================================================================== -->
## Export and import user flows as files

> To share user flows with other people, for example to help other people reproduce a bug by running the same set of steps as you, export the user flows you recorded as files.

Reviewer: ```suggestion
To share user flows with other people, for example to help other people reproduce a bug by running the same set of steps as you, export the user flows that you recorded, as files.
```
